includes/Core/Permissions/Permissions.php

```diff
@@ -93,31 +114,48 @@ final class Permissions {
 	 *
 	 * @param Context        $context        Plugin context.
 	 * @param Authentication $authentication Optional. Authentication instance. Default is a new instance.
+	 * @param Modules        $modules        Optional. Modules instance. Default is a new instance.
 	 */
-	public function __construct( Context $context, Authentication $authentication = null ) {
+	public function __construct( Context $context, Authentication $authentication = null, Modules $modules = null ) {
 		$this->context = $context;
 
 		if ( ! $authentication ) {
 			$authentication = new Authentication( $this->context );
 		}
 		$this->authentication = $authentication;
 
+		if ( ! $modules ) {
+			$modules = new Modules( $this->context, null, null, $this->authentication );
+		}
+		$this->modules = $modules;
+
+		// TODO Remove the temporary assignment of these capabilities when Dashboard Sharing feature flag is removed.
+		$editor_capability        = 'manage_options';
+		$admin_network_capability = 'manage_options';
+		if ( Feature_Flags::enabled( 'dashboardSharing' ) ) {
+			$editor_capability        = 'edit_posts';
+			$admin_network_capability = 'manage_network';
+		}
+
 		$this->base_to_core = array(
 			// By default, only allow administrators to authenticate.
-			self::AUTHENTICATE        => 'manage_options',
+			self::AUTHENTICATE          => 'manage_options',
 
 			// Allow contributors and up to view their own post's insights.
-			// TODO change to map to edit_posts when Site Kit supports non admin access.
-			self::VIEW_POSTS_INSIGHTS => 'manage_options',
+			// TODO change to map to edit_posts when Dashboard Sharing feature flag is removed.
+			self::VIEW_POSTS_INSIGHTS   => $editor_capability,
 
 			// Allow editors and up to view the dashboard and module details.
-			// TODO change to map to edit_others_posts when Site Kit supports non admin access.
-			self::VIEW_DASHBOARD      => 'manage_options',
-			self::VIEW_MODULE_DETAILS => 'manage_options',
+			// TODO change to map to edit_posts when Dashboard Sharing feature flag is removed.
+			self::VIEW_DASHBOARD        => $editor_capability,
+			self::VIEW_MODULE_DETAILS   => $editor_capability,
+
+			// Allow editors and up to view shared dashboard data.
+			self::VIEW_SHARED_DASHBOARD => 'edit_posts',
```

Comment:
Well I assumed this to be the case based on @aaemnnosttv 's [comment](https://github.com/google/site-kit-wp/issues/4522#issuecomment-1015298695). He pointed that I was supposed to use 'edit_posts' rather than 'edit_others_posts' as the core primitive capability to map to. Apologies I should've updated the other snippet in the IB - but since I didn't could you verify this looks good @aaemnnosttv?

Reply:
@jimmymadon @techanvil `edit_posts` is correct. I believe the `edit_others_posts` was from an old TODO comment before dashboard sharing was designed.

Reply:
Cool, thanks for clarifying @jimmymadon @aaemnnosttv :+1: